src/particle.f90

```diff
@@ -0,0 +1,2187 @@
+    real(mytype),intent(in) :: particle_range(6)
+    !
+    ! local data
+    integer :: p,i,j,k,n,local_size,code,npx,npy,npz
+    integer :: big_number
+    integer,allocatable :: seed(:)
+    real(mytype) :: x,y,z,detalx,detaly,detalz
+    real(mytype) :: plx,ply,plz
+    !
+
+    plx=particle_range(2)-particle_range(1)
+    ply=particle_range(4)-particle_range(3)
+    plz=particle_range(6)-particle_range(5)
+
+    call particle_dis_xyz(n_particles,plx,ply,plz,npx,npy,npz)
+    !
+    if(nrank==0) then
+        print*,'** actual number of particle in 3 direction',npx,npy,npz,'=',npx*npy*npz
+    endif
+
+    detalx = (particle_range(2)-particle_range(1))/real(npx,mytype)
+    detaly = (particle_range(4)-particle_range(3))/real(npy,mytype)
+    detalz = (particle_range(6)-particle_range(5))/real(npz,mytype)
+    !
+    big_number=npx*npy*npz
+
+    allocate(particle_new(1:big_number))
+    !
+    p=0
+    do k=1,npz
+    do j=1,npy
+    do i=1,npx
+      !
+      x=particle_range(1)+0.5_mytype*detalx+detalx*real(i-1,mytype)
+      y=particle_range(3)+0.5_mytype*detaly+detaly*real(j-1,mytype)
+      z=particle_range(5)+0.5_mytype*detalz+detalz*real(k-1,mytype)
+      !
+      if( x>=lxmin(nrank) .and. x<lxmax(nrank) .and. &
+          y>=lymin(nrank) .and. y<lymax(nrank) .and. &
+          z>=lzmin(nrank) .and. z<lzmax(nrank) ) then
+        !
+        p=p+1
+        !
+        call particle_new(p)%init()
+        !
+        particle_new(p)%x(1) = x
+        particle_new(p)%x(2) = y
+        particle_new(p)%x(3) = z
+        !
+        particle_new(p)%v(1) = 0._mytype
+        particle_new(p)%v(2) = 0._mytype
+        particle_new(p)%v(3) = 0._mytype
+        !
+        particle_new(p)%new=.false.
+        !
+      endif
+      !
+    enddo
+    enddo
+    enddo
+    !
+    call mclean(particle_new,p)
+    !
+    particle_size=p
+
+  end subroutine particle_gen_uniform
+  !+-------------------------------------------------------------------+
+  !| The end of the subroutine particle_gen.                           |
+  !+-------------------------------------------------------------------+
+
+
+  !+-------------------------------------------------------------------+
+  !| This subroutine is to distribute particles evenly in 3 directions |
+  !+-------------------------------------------------------------------+
+  !| CHANGE RECORD                                                     |
+  !| -------------                                                     |
+  !| 17-10-2024  | Created by J. Fang                                  |
+  !+-------------------------------------------------------------------+
+  subroutine particle_dis_xyz(nptotal,plx,ply,plz,npx,npy,npz)
+
+    ! arguments
+    integer,intent(in) ::  nptotal
+    real(mytype),intent(in) :: plx,ply,plz
+    integer,intent(out) ::  npx,npy,npz
+
+    ! local data
+    real(mytype) :: n1
+
+
+    if(abs(plx)<1.d-16 .and. abs(ply)<1.d-16 .and. abs(plz)<1.d-16) then
```

Comment:
replace `1.d-16` with `epsilon(1.0_mytype)`. I would add this to the local data of subroutine, e.g.
```
real(mytype), parameter :: eps = epsilon(1.0_mytyp)
```
or you could make this a module variable if used elsewhere.

Reply:
applied.